packages/terminal/src/browser/terminal-linkmatcher-files.ts

```diff
@@ -0,0 +1,172 @@
+        return async (match) => {
+            const toOpen = await this.toURI(match, await terminalWidget.cwd);
+            if (toOpen) {
+                // TODO: would be better to ask the opener service, but it returns positively even for unknown files.
+                const f = await this.fileSystem.getFileStat(toOpen.toString());
+                return f !== undefined && !f.isDirectory;
+            }
+            return false;
+        };
+    }
+
+    getHandler(terminalWidget: TerminalWidgetImpl): (event: MouseEvent, link: string) => void {
+        return async (event, fullMatch) => {
+            const toOpen = await this.toURI(fullMatch, await terminalWidget.cwd);
+            if (!toOpen) {
+                return;
+            }
+            const position = await this.extractPosition(fullMatch);
+            let options = {};
+            if (position) {
+                options = {
+                    selection: {
+                        start: position
+                    }
+                };
+            }
+            try {
+                const opener = await this.openerService.getOpener(toOpen, options);
+                opener.open(toOpen, options);
+            } catch (err) {
+                console.error('Cannot open link ' + fullMatch, err);
+            }
+        };
+    }
+
+    protected async toURI(match: string, cwd: URI): Promise<URI | undefined> {
+        const path = await this.extractPath(match);
+        if (!path) {
+            return;
+        }
+        const pathObj = new Path(path);
+        return pathObj.isAbsolute ? cwd.withPath(path) : cwd.resolve(path);
+    }
+
+    protected async extractPosition(link: string): Promise<Position> {
+        const matches: string[] | null = (await this.getRegExp()).exec(link);
+        const info: Position = {
+            line: 1,
+            character: 1
+        };
+
+        if (!matches) {
+            return info;
+        }
+
+        const lineAndColumnMatchIndex = (await this.backendOs) === OS.Type.Windows ? winLineAndColumnMatchIndex : unixLineAndColumnMatchIndex;
+        for (let i = 0; i < lineAndColumnClause.length; i++) {
+            const lineMatchIndex = lineAndColumnMatchIndex + (lineAndColumnClauseGroupCount * i);
+            const rowNumber = matches[lineMatchIndex];
+            if (rowNumber) {
+                info.line = parseInt(rowNumber, 10) - 1;
+                const columnNumber = matches[lineMatchIndex + 2];
+                if (columnNumber) {
+                    info.character = parseInt(columnNumber, 10) - 1;
+                }
+                break;
+            }
+        }
+
+        return info;
+    }
+
+    protected async extractPath(link: string): Promise<string | undefined> {
+        const matches: string[] | null = (await this.getRegExp()).exec(link);
+        if (!matches) {
+            return undefined;
+        }
+        return matches[1];
+    }
+}
+
+// below reg exps are taken from
```

Comment:
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19821

Reply:
@svenefftinge, I think we need to attach an archive containing the specific revision of the file copied from and the LICENSE.
If you want, I'll add it to the CQ

Reply:
Ok, thank you.